### PR TITLE
Support fold batch norm for atrous conv2d

### PR DESCRIPTION
**Pull request by liyinhgqw**
Fix #13990

As we can fold batch norm with convolution, we should also fold batch norm with atrous convolution, which has not been implemented.

### CODE REVIEW

**Comment by martinwicke** [comment]
@petewarden could you take a look or assign someone who can?

**Review by petewarden — COMMENTED** [review]
We're trying to encourage people to migrate to the Graph Transform Tool:
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/graph_transforms/#optimizing-for-deployment

If possible, it would be great if you could make this change in that path too, probably in this file (and the related tests):

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/graph_transforms/fold_batch_norms.cc

Let me know if that's possible, thanks for the contribution!

**Comment by liyinhgqw** [comment]
I updated the the Graph Transform Tool to fold batchnorm with atrous convolution.
And add unit-test for it.

**Comment by martinwicke** [comment]
Ping @petewarden

**Comment by drpngx** [comment]
@petewarden any luck with this?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by petewarden — APPROVED** [review]
Sorry for my very slow response on this one, but thanks for the fix, this looks good!

**Comment by drpngx** [comment]
The test failed (`TestFoldFusedBatchNormsWithBatchToSpace`):
```
tensorflow/tools/graph_transforms/fold_old_batch_norms_test.cc:356
      Expected: ::tensorflow::Status::OK()
      Which is: OK
To be equal to: (root.ToGraphDef(&original_graph_def))
      Which is: Invalid argument: Shapes must be equal rank, but are 1 and 2 for 'batch_to_space_op' (op: 'BatchToSpaceND') with input shapes: [1,1,5,2], [2], [2].
```

**Comment by martinwicke** [comment]
ping @liyinhgqw, could you look at the failing test?

**Comment by protoget** [comment]
ping @liyinhgqw, could you look at the failing test?

**Comment by liyinhgqw** [comment]
okay. I'll check it out.

**Comment by liyinhgqw** [comment]
@petewarden @drpngx @martinwicke @protoget , I fixed some test data. Could you please check it again?
